cobra/core/Metabolite.py

```diff
@@ -41,48 +43,58 @@ def __init__(self, id=None, formula=None,
         self._bound = 0.
 
     @property
+    def formula(self):
```

Comment:
I considered doing this, but opted not to for performance reasons. This means that every time a metabolite is created, this check must run.

If you want the formula getter/setter let's do it in a separate PR. We'd have to do some benchmarking on it to ensure the performance isn't too bad, and probably switch to using metabolite._formula in some parts of the code (such as I/O)